runtime/syntax/gleam.yaml

```diff
@@ -0,0 +1,36 @@
+filetype: gleam
+
+detect:
+    filename: "\\.gleam$"
+
+rules:
+    - identifier: "\\b[a-z][a-z0-9_]*\\b"
+    - ignore: "(\\b(_[a-z][a-z0-9_]*)\\b|\\b(auto|delegate|derive|echo|else|implement|macro|test)\\b)"
+    - identifier.class: "[a-z][a-z0-9_]*\\("
+    - statement: "\\b(import|as|assert|case|const|if|let|panic|todo|use|opaque|pub|type|fn)\\b"
+    - type: "\\b[A-Z]+[A-Za-z0-9_]*\\b"
+    - constant.string:
+        start: '"'
+        end: '"'
+        skip: '\\"'
+        rules: []
+    # bit arrays
+    - special:
+        start: "<<"
+        end: ">>"
+        rules: []
```

Comment:
I actually thought it might be clearer as is, making them look similar to strings, due to their special syntax. I took an initial stab, but a bug in the regex helps make a comparison. Here, I have no idea what makes `:size(16)` not match `:(size|unit|bits|bytes|float|int|big|little|native|utf8|utf16|utf32|utf8_codepoint|utf16_codepoint|utf32_codepoint|signed|unsigned)(\\([0-9]+\\))?`, but I actually think having the entire bit array highlighted similar to a string makes it read easier...

![image](https://github.com/zyedidia/micro/assets/1424402/d82d78a1-7bb5-4e15-a630-541b5e8a3d29)